tests/extensions/test_table.py

```diff
@@ -59,3 +59,21 @@ def test_should_raise_exception_when_passing_invalid_extension_object(
             TableExtension.ext,
             object(),
         )
+
+    def test_item_with_table_extension_is_serilalizable_and_roundtrips(self) -> None:
+        item = pystac.Item.from_file(self.example_uri)
+        # add column metadata
+        tab_ext = TableExtension.ext(item, add_if_missing=True)
+        columns = [
+            Column({"name": "col_1", "type": "str"}),
+            Column({"name": "col_2", "type": "byte_array"}),
+        ]
+        tab_ext.columns = columns
+        item.save_object()
+        self.assertTrue(all(isinstance(c, Column) for c in tab_ext.columns))
+        self.assertTrue(
+            all(
+                before.properties == after.properties
+                for before, after in zip(columns, tab_ext.columns)
+            )
+        )
```

Comment:
```suggestion
 def test_item_with_table_extension_is_serilalizable_and_roundtrips() -> None:
      item = pystac.Item.from_file(self.example_uri)
      # add column metadata
      tab_ext = TableExtension.ext(item, add_if_missing=True)
      columns = [
          Column({"name": "col_1", "type": "str"}),
          Column({"name": "col_2", "type": "byte_array"}),
      ]
      tab_ext.columns = columns
      item.save_object()
      assert all(isinstance(c, Column) for c in tab_ext.columns)
      assert (
          all(
              before.properties == after.properties
              for before, after in zip(columns, tab_ext.columns)
          )
      )
```

Since #939 we have **pytest**; while we haven't backported our existing tests away from `unittest.TestCase` format yet, I think it's worth starting to define new tests as free functions to reduce our future work. I've opened https://github.com/stac-utils/pystac/issues/993 to capture the need to switch over the whole test suite (no small thing).

Reply:
:100: Oh thank god. I am way more comfortable in pytest.